cookbooks/bcpc/libraries/utils.rb

```diff
@@ -67,11 +78,26 @@ def make_config(key, value)
 	end
 end
 
-def get_config(key)
-	init_config if $dbi.nil?
-	Chef::Log.info  "------------ Fetching value for key \"#{key}\""
-	value = node['bcpc']['encrypt_data_bag'] ? $edbi[key] : $dbi[key]
-	return value
+# get value for data bag/chef-vault item with key
+# bag: databag name of chef vault item; 
+# item: chef vault item; 
+# key: the key to retrieve password in the item
+def get_config(bag="configs",item=node.chef_environment,key)
+  require 'chef-vault'
+  begin
+    entry = ChefVault::Item.load(bag, item)
+    return entry[key]
+  rescue ChefVault::Exceptions::KeysNotFound
+    if bag != "configs"
```

Comment:
In the `rescue` block we should only keep following two lines:
````
entry = Chef::DataBagItem.load(bag,item)
return entry[key]
````

Everything else can go away.